packages/stepper/src/Stepper.tsx

```diff
@@ -1,331 +1,107 @@
-import CheckmarkIcon from '@leafygreen-ui/icon/dist/Checkmark';
-import EllipsisIcon from '@leafygreen-ui/icon/dist/Ellipsis';
-import {
-  completedStepLabelStyle,
-  containerStyle,
-  currentStepLabelStyle,
-  currentStepStyle,
-  dataProps,
-  hiddenLayerStyle,
-  layerStyle,
-  previewStyle,
-  slideInAnimationStyle,
-  slideOutAnimationStyle,
-  stepBoxBottomStyle,
-  stepBoxStyle,
-  stepBoxTopStyle,
-  stepLabelStyle,
-  stepLabelTextStyle,
-  stepStyle,
-  stepTextStyle,
-  upcomingStepLabelStyle,
-  upcomingStepStyle,
-} from './styles';
-
-type StepCompletionState = 'completed' | 'current' | 'upcoming';
-
-type InternalStepProps = JSX.IntrinsicElements['div'] & {
-  children: React.ReactNode;
-  state: StepCompletionState;
-  stepLabel?: string;
-  isPreview: boolean;
-};
-
-function InternalStep({
-  children,
-  state,
-  stepLabel,
-  isPreview,
-  className,
-  ...containerProps
-}: InternalStepProps) {
-  const isCurrent = state === 'current';
-  const isUpcoming = state === 'upcoming';
-  const isCompleted = state === 'completed';
-  return (
-    <div
-      className={`${cx(stepStyle, {
-        [currentStepStyle]: isCurrent,
-        [upcomingStepStyle]: isUpcoming,
-        [previewStyle]: isPreview,
-      })} ${className}`}
-      {...containerProps}
-    >
-      {stepLabel && (
-        <div
-          className={cx(stepLabelStyle, {
-            [currentStepLabelStyle]: isCurrent,
-            [completedStepLabelStyle]: isCompleted,
-            [upcomingStepLabelStyle]: isUpcoming,
-            [previewStyle]: isPreview,
-          })}
-        >
-          {isCompleted ? (
-            <CheckmarkIcon />
-          ) : (
-            <div className={stepLabelTextStyle}>{stepLabel}</div>
-          )}
-        </div>
-      )}
-      <div className={cx(stepTextStyle, { [previewStyle]: isPreview })}>
-        {children}
-      </div>
-      <div className={stepBoxStyle}>
-        <span className={stepBoxTopStyle} {...dataProps.stepBoxTop.prop}>
-          <div />
-        </span>
-        <span className={stepBoxBottomStyle} {...dataProps.stepBoxBottom.prop}>
-          <div />
-        </span>
-      </div>
-    </div>
-  );
-}
-
-InternalStep.displayName = 'Step';
-
-type StepItem = InternalStepProps & {
-  step: 'previous' | 'next' | number;
-};
-
-type StepContents = string | number | Array<StepContents>;
-
-type StepElement = React.ReactElement<{ children: StepContents }>;
-export type StepElements = StepElement | Array<StepElement>;
-
-interface StepperProps {
-  children: StepElements;
-  currentStep: number;
-  maxDisplayedSteps?: number;
-  className?: string;
-}
-
-export function Stepper({
+import React, { PropsWithChildren } from 'react';
+import { css, cx } from '@leafygreen-ui/emotion';
+import Step from './InternalStep';
+import { StepStates, StepperProps } from './types';
+import EllipsesStep from './EllipsesStep';
+import StepperContextProvider from './StepperContext';
+
+const Stepper = ({
```

Comment:
Is there a way to add prop validations to check values of props against other props? I'm not personally familiar. 

We could definitely have our own checks to throw our own errors though. I didn't play around with that idea because the previous version of Stepper didn't check for values.

Reply:
You could get fancy with Typescript prop types, but I'd just recommend a check in the render function that warns in the console (`lib.consoleOnce.warn`) when the props don't make sense

Reply:
As discussed, because it's a good amount of extra effort and maintains parity with the pre-redesign Stepper, this work has been added to the backlog here: https://jira.mongodb.org/browse/PD-1964